overwatch/src/overwatch/handle.rs

```diff
@@ -37,14 +42,33 @@ impl OverwatchHandle {
         }
     }
 
-    #[must_use]
-    /// Request a relay
-    pub fn relay<Service>(&self) -> Relay<Service>
+    /// Request a relay with a service
+    pub async fn relay<Service>(&self) -> Result<OutboundRelay<Service::Message>, RelayError>
     where
         Service: ServiceData,
         Service::Message: 'static,
     {
-        Relay::new(self.clone())
+        info!("Requesting relay with {}", Service::SERVICE_ID);
+        let (sender, receiver) = tokio::sync::oneshot::channel();
+        let Ok(()) = self
+            .send(OverwatchCommand::Relay(RelayCommand {
+                service_id: Service::SERVICE_ID,
+                reply_channel: ReplyChannel::from(sender),
+            }))
+            .await
+        else {
+            unreachable!("Service relay should always be available");
+        };
+        let message = receiver
+            .await
+            .map_err(|e| RelayError::Receiver(Box::new(e)))??;
+        message
+            .downcast::<OutboundRelay<Service::Message>>()
+            .map(|r| *r)
+            .map_err(|m| RelayError::InvalidMessage {
+                type_id: format!("{:?}", (*m).type_id()),
+                service_id: Service::SERVICE_ID,
+            })
```

Comment:
I wonder if this could be simplified now. Should a `relay` fail? I mean, whos responsibility is if it fails? If it is overwatch its probably because we did something wrong and should be a panic (?)

Reply:
Yeah to be honest, now that I think a bit more deeply about it, I think that such a foundational library should never have explicit panics or expects, but all errors should be propagated to the caller. It's already fine to panic in nomos, but here I feel we can do better and avoid panics and unwraps altogether. What do you think? I can update all the code to avoid panics and return errors everywhere.

Reply:
I followed the same approach used for the settings update flow: https://github.com/logos-co/Overwatch/pull/60/commits/156ff32873521a14fa596904bd27c04525a355ca.

Reply:
I think panic here and there it is fine. But just in specific cases. In this case, panic means that you couldn't relay. Even if we bubble up an error, the final user of the library cannot do anything about it. It is a fundamental piece that **should always work** IMO. If it doesn't it is because *we* (as lib devs) probably did something wrong.